Sources/LeafErrorMiddleware/LeafErrorMiddleware.swift

```diff
@@ -1,61 +1,76 @@
         self.environment = environment
-        self.viewRenderer = viewRenderer
+        self.log = log
     }
 
-    public func respond(to req: Request, chainingTo next: Responder) throws -> Response {
-        do {
-            return try next.respond(to: req)
-        } catch {
-            log.error(error)
-            return make(with: req, for: error)
-        }
-    }
+    /// See `Middleware.respond`
+    public func respond(to req: Request, chainingTo next: Responder) throws -> Future<Response> {
+        let renderer = try req.make(LeafRenderer.self)
+
+        func handleError(with status: HTTPStatus) throws -> Future<Response> {
+            if status == .notFound {
+                do {
+                    return try renderer
+                        .render("404")
+                        .encode(for: req)
+                        .map(to: Response.self) { res in
+                            res.http.status = status
+                            return res
+                    }
+                } catch { /* swallow so we return the default view */ }
+            }
 
-    public func make(with req: Request, for error: Error) -> Response {
-        let status: Status = Status(error)
-        if status == .notFound {
             do {
-                let response = try viewRenderer.make("404", Node([:])).makeResponse()
-                response.status = .notFound
-                return response
-            } catch { /* swallow so we return the default view */ }
+                let parameters: [String:String] = [
+                    "status": status.code.description,
+                    "statusMessage": status.reasonPhrase
+                ]
+                return try renderer
+                    .render("serverError", parameters)
+                    .encode(for: req)
+                    .map(to: Response.self) { res in
+                        res.http.status = status
+                        return res
+                }
+            } catch {
+                let body = "<h1>Internal Error</h1><p>There was an internal error. Please try again later.</p>"
+                return try body.encode(for: req)
+                    .map(to: Response.self) { res in
+                        res.http.status = status
+                        res.http.headers.replaceOrAdd(name: .contentType, value: "text/html; charset=utf-8")
+                        return res
+                }
+            }
         }
-        
+
         do {
-            let parameters: [String: NodeRepresentable] = [
-                "status": status.statusCode.description,
-                "statusMessage": status.reasonPhrase
-            ]
-            let response = try viewRenderer.make("serverError", parameters).makeResponse()
-            response.status = status
-            return response
+            return try next.respond(to: req)
+                .flatMap(to: Response.self) { res in
+                    if res.http.status != .ok {
```

Comment:
Not sure how useful this would turn out to be in real life. Maybe `if res.http.status.code >= HTTPResponseStatus.badRequest.code` instead